my_game.py

```diff
@@ -419,132 +430,100 @@ def setup(self):
-            )
+        # and their shots
+        self.ufo_shot_list.draw()
 
-        # only post-game
-        elif self.game_state == GameState.GAME_OVER:
-            pass
+        # Draw players score on screen
+        arcade.draw_text(
+            "SCORE: {}".format(self.player_score),  # Text to show
+            10,  # X position
+            SCREEN_HEIGHT - 20,  # Y_position
+            arcade.color.WHITE  # Color of text
+        )
+        arcade.draw_text(
+            "LIVES: {}".format(self.player_sprite.lives),  # Text to show
+            10,  # X position
+            SCREEN_HEIGHT - 45, # Y positon
+            arcade.color.WHITE  # Color of text
+        )
 
     def on_update(self, delta_time):
         """
         Movement and game logic
         """
 
-        if self.game_state == GameState.IN_GAME:
-            # Calculate player speed based on the keys pressed
-            # Move player with keyboard
-            if self.left_pressed and not self.right_pressed:
-                self.player_sprite.angle+= PLAYER_ROTATE_SPEED
-            elif self.right_pressed and not self.left_pressed:
-                self.player_sprite.angle+= -PLAYER_ROTATE_SPEED
-
-            # rotate player with joystick if present
-            if self.joystick:
-                self.player_sprite.angle += round(self.joystick.x) * -PLAYER_ROTATE_SPEED
-
-            # checks if ufo shot collides with player
-            if any(self.player_sprite.collides_with_list(self.ufo_shot_list)):
-                self.player_sprite.lives -= 1
-
-            #player shot
-            for shot in self.player_shot_list:
-
-                for ufo_hit in arcade.check_for_collision_with_list(shot, self.ufo_list):
-                    shot.kill()
-                    ufo_hit.destroy()
-                    self.player_score += UFO_POINTS_REWARD
-                
-            # Check for PlayerShot - Asteroid collisions
-            for s in self.player_shot_list:
-                for a in arcade.check_for_collision_with_list(s, self.asteroid_list):
-                    s.kill()
-                    a.kill()
-
-            # check for thrust
-            if self.thrust_pressed:
-                self.player_sprite.thrust()
-
-            # Update player sprite
-            self.player_sprite.update()
+        # Calculate player speed based on the keys pressed
+        # Move player with keyboard
+        if self.left_pressed and not self.right_pressed:
+            self.player_sprite.angle+= PLAYER_ROTATE_SPEED
+        elif self.right_pressed and not self.left_pressed:
+            self.player_sprite.angle+= -PLAYER_ROTATE_SPEED
+
+        # rotate player with joystick if present
+        if self.joystick:
+            self.player_sprite.angle += round(self.joystick.x) * -PLAYER_ROTATE_SPEED
+
+        # checks if ufo shot collides with player
+        if any(self.player_sprite.collides_with_list(self.ufo_shot_list)):
+            self.player_sprite.lives -= 1
+
```

Comment:
Will the player not be hit by the same ufo shot in the next frame?

Reply:
it will. Gonna fix that right after this.

Reply:
Was fixed in other branch.